### PR TITLE
feat: add scripts that on release will publish transpiled template to npm

### LINKED ISSUE #558: Add necessary scripts that on release will publish transpiled template to npm
Since we migrated this template to React render engine, the generation time, like in other react templates, got super long

Reason - every time generation is run, TS transpilation runs

This can be solved by publishing templates, like this one, to npm with transpilation files included, so basically no transpilation is needed in runtime - unless explicitly requested.

When you clone this repo, install and run `asyncapi generate fromTemplate ./test/spec/asyncapi_v3.yml ./ --output mydocs --force-write` from root, you will notice a new folder called `__transpiled` - this is the folder we need to make sure is included in package publishing.

This is how generator triggers transpilation -> https://github.com/asyncapi/generator/blob/master/lib/renderer/react.js#L19
Now we need to make sure we have proper script here in template that does the same. This new script needs to be called only by this script -> https://github.com/asyncapi/html-template/blob/master/package.json#L33C6-L33C20

We already have some custom scripts in this repo that run on release -> https://github.com/asyncapi/html-template/blob/master/scripts/copy-sources.js so you just plugin into existing infra.

**Note**
Completion of this task do not mean yet that time of generation will go down. I think that babel that we use in https://github.com/asyncapi/generator-react-sdk/blob/master/src/transpiler/transpiler.ts is already smart and do not regenerate if `__transpiled` dir is already there but yeah, lets see

### PR DESCRIPTION
**Pull request by utnim2**
**Description**
- Added the transpile scripts that creates `__transpiled ` folder in the root directory
- Modify the `prepublishOnly` scripts that ensures transpilation runs before publishing

**Related issue(s)**
Fixes #558

### CODE REVIEW

File: scripts/transpile.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+const { transpileFiles } = require('@asyncapi/generator-react-sdk');
+const path = require('path')
+
+async function main(){
```
Comment (derberg) [inline scripts/transpile.js:4]: I don't think it's needed to wrap it like this in "main" function cause transpileTemplate is a wrapper already

File: scripts/transpile.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  async function transpileTemplate() {
+    const templateContentDir = path.join(__dirname, '../template');
+    console.log("templateContentDir",templateContentDir)
```
Comment (derberg) [inline scripts/transpile.js:7]: you can put nicer names for these logs

**Review by derberg — COMMENTED** [review]
I don't think we actually need to modify copy sources script as we in the end publish entire package, not only template folder. Copy sources is needed only for specific thing.

So yeah, package.json script modification should be enough. Only one more change is needed - _trabspiled is in gitignore, so by default will not be pushed to npm,so you need to modify package.json so it accepts _trabspile folder

File: scripts/transpile.js (hunk 2 above)
Comment (utnim2) [reply scripts/transpile.js:7]: 😅

File: scripts/transpile.js (hunk 1 above)
Comment (utnim2) [reply scripts/transpile.js:4]: done 👍

File: scripts/transpile.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+const { transpileFiles } = require('@asyncapi/generator-react-sdk');
+const path = require('path')
+
+async function transpileTemplate() {
+  const templateContentDir = path.join(__dirname, '../template');
+  console.log("templateContentDir",templateContentDir)
```
Comment (derberg) [inline scripts/transpile.js:6]: please add full sentence console log

File: scripts/transpile.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+  const outputDir = path.join(__dirname, '../__transpiled');
```
Comment (derberg) [inline scripts/transpile.js:7]: console log this variable as well, but again, with full descriptive console log info

File: scripts/transpile.js [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+  await transpileFiles(templateContentDir, outputDir, {recursive: true})
```
Comment (derberg) [inline scripts/transpile.js:8]: please put it in between try/catch instead of later catching errors on transpileTemplate

**Review by derberg — COMMENTED** [review]
looks good, just some code improvements please

**Comment by utnim2** [comment]
Done @derberg 👍

**Review by derberg — APPROVED** [review]
LGTM, lets merge and see if transpiled files end up in a package and if it makes any difference

**Comment by aeworxet** [comment]
@asyncapi/bounty_team

**Pull request merged**